geoh5py/objects/drillhole.py

```diff
@@ -733,60 +721,136 @@ def post_processing(self):
+                ]
+            )
 
-    :return deviation: Change in northing distance.
-    """
-    return np.sin(np.deg2rad(450.0 - azimuth % 360.0)) * np.cos(np.deg2rad(dip))
+        if len(new_points) > 1:
+            full_survey = np.vstack([full_survey, np.vstack(new_points)])
+
+        return np.unique(full_survey, axis=0)
+
+    @staticmethod
+    def compute_intervals(survey: np.ndarray, collar, end_of_hole) -> dict:
+        """
+        Compute the intervals from survey and collar information.
+
+        Stores a dictionary describing the arc circle between the survey points
+        separated by a maximum depth interval of 50 m.
+
+        Translated from C++ code.
+
+        :param survey: Array of survey points with columns 'Depth', 'Azimuth', 'Dip'.
+        :param collar: Collar location of the drillhole.
+        :param end_of_hole: End of the drillhole in meters.
+        """
+        full_survey = Drillhole.validate_survey_values(survey, end_of_hole)
+        unit_vector = dip_azimuth_to_vector(full_survey[:, 2], full_survey[:, 1])
+
+        if survey.shape[0] < 2:
+            return {
+                "depths": np.r_[0],
+                "rad": np.r_[INFINITE_RADIUS],
+                "tangential": np.zeros((1, 3)),
+                "unit_vector": unit_vector,
+                "locations": np.c_[collar],
+            }
+
+        # Cross product between neighbours
+        cross = np.cross(unit_vector[:-1, :], unit_vector[1:, :], axis=1)
+        vr = np.linalg.norm(cross, axis=1)
+        dot = np.sum(unit_vector[:-1, :] * unit_vector[1:, :], axis=1)
+        tangential = unit_vector[1:, :] - dot[:, None] * unit_vector[:-1, :]
+        norm = np.linalg.norm(tangential, axis=1)
+        norm[norm == 0.0] = INFINITE_RADIUS
+        tangential /= norm[:, None]
+        alpha = np.abs(0.5 * np.pi - np.arctan2(dot, vr))
+        delta_depth = np.diff(full_survey[:, 0])
+        radius = delta_depth / alpha
+
+        radius[alpha == 0.0] = delta_depth[alpha == 0.0] * INFINITE_RADIUS
+        alpha[alpha == 0.0] = delta_depth[alpha == 0.0] / radius[alpha == 0.0]
+
+        intervals = {
+            "depths": np.r_[full_survey[:, 0]],
+            "rad": np.r_[radius, INFINITE_RADIUS],
+            "tangential": np.vstack([tangential, np.zeros(3)]),
+            "unit_vector": unit_vector,
+            "locations": np.r_[
+                np.c_[collar],
+                collar
+                + np.cumsum(
+                    radius[:, None]
+                    * (
+                        np.sin(alpha)[:, None] * unit_vector[:-1, :]
+                        + (1 - np.cos(alpha))[:, None] * tangential
+                    ),
+                    axis=0,
+                ),
+            ],
+        }
 
+        return intervals
 
-def deviation_z(_, dip):
-    """
-    Compute the vertical deviation.
+    @staticmethod
+    def depths_to_xyz(intervals: dict, depths: np.ndarray) -> np.ndarray:

```

Comment:
why intervals is not loaded from self? (why static?)

the idea of depth_to_xyz is not to do a desurvey?

like drillholes.desurvey(depth_values) => xyz coords?

Reply:
The reason being that the methods can be used outside of an instance of Drillhole, like what the GI team needs.